src/components/Dialog.tsx

```diff
@@ -8,296 +8,350 @@ import {
       nacelleSpaceToken: token,
       nacelleEndpoint: endpoint,
-    } = this.props.sdk.parameters.installation as AppInstallationParameters;
-    const invocation = this.props.sdk.parameters.invocation as DialogState;
-    const { value } = invocation;
+    } = this.props.sdk.parameters.installation as AppInstallationParameters
+    const invocation = this.props.sdk.parameters.invocation as DialogState
+    const { value } = invocation
 
     // Set data handlers
-    await this.setClient(id, token, endpoint);
-    await this.setW2Settings(id, token, endpoint);
-    await this.setStorage(id);
+    await this.setClient(id, token, endpoint)
+    await this.setW2Settings(id, token, endpoint)
+    await this.setStorage(id)
 
-    if (endpoint && endpoint.includes("storefront")) {
-      this.setState(() => ({ isW2: true }));
+    if (endpoint && endpoint.includes('storefront')) {
+      this.setState(() => ({ isW2: true }))
     }
 
     // Get resources from invocation
-    await this.refreshData(false);
+    await this.refreshData(false)
 
     this.setState((state) => ({
       ...invocation,
       publishedValue: value,
       loading: false,
-    }));
+    }))
   }
 
   setW2Settings = async (id: string, token: string, endpoint: string) => {
     const w2Settings = {
       id,
       token,
-      locale: "en-us",
+      locale: 'en-us',
       nacelleEndpoint: endpoint,
-    };
+    }
 
-    this.setState(() => ({ w2Settings }));
-  };
+    this.setState(() => ({ w2Settings }))
+  }
 
   setClient = async (id: string, token: string, endpoint: string) => {
     const client = new NacelleClient({
       id,
       token,
-      locale: "en-us",
-      nacelleEndpoint: endpoint || "https://hailfrequency.com/v2/graphql",
+      locale: 'en-us',
+      nacelleEndpoint: endpoint || 'https://hailfrequency.com/v2/graphql',
       useStatic: false,
       disableEvents: true,
-    });
+    })
 
-    this.setState(() => ({ client }));
-  };
+    this.setState(() => ({ client }))
+  }
 
   setStorage = async (name: string) => {
     // Namespace storage with Space Id to avoid conflict if managing multiple spaces.
-    const storage = localforage.createInstance({ name });
-    this.setState(() => ({ storage }));
-  };
+    const storage = localforage.createInstance({ name })
+    this.setState(() => ({ storage }))
+  }
 
   refreshData = async (force: boolean = false) => {
     if (force) {
-      await this.state.storage.removeItem("products-expires");
-      await this.state.storage.removeItem("collections-expires");
+      await this.state.storage.removeItem('products-expires')
+      await this.state.storage.removeItem('collections-expires')
     }
-    await this.fetchResources("products");
-    await this.fetchResources("collections");
-    await this.setSelectedTab(this.state.selectedTabId);
-  };
+    await this.fetchResources('products')
+    await this.fetchResources('collections')
+    await this.setSelectedTab(this.state.selectedTabId)
```

Comment:
`this.setSelectedTab` is not async.

```suggestion
    this.setSelectedTab(this.state.selectedTabId)
```